extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/TracerConfig.java

```diff
@@ -19,8 +20,52 @@
     /** Build / static runtime config for span exporters */
     public SpanExporterConfig exporter;
 
+    /**
+     * Tracing ID generator.
+     * <p>
+     * Valid value is {@code xray}.
+     * <p>
+     * If not set defaults to the default OpenTelemetry ID Generator.
+     */
+    @ConfigItem
```

Comment:
I can't remember what I'd said about this before, but wondering if we need configuration for it.

We could have the user creating a CDI bean of `IdGenerator`, which the recorder picks up, if there is one, or uses the default otherwise.

Reply:
Before we had a default bean, and I removed the default bean and added the config param that if not set it does nothing and if set it sets the configured id generator, the only difference is that we are not using CDI whatsoever

Reply:
Have you verified this works in an app not using AWS extensions?

My concern is that the dependency on AWS, and others, marked `provided` will cause a runtime failure when executing without those dependencies present.

If we dropped the config, but looked for an `IdGenerator` CDI bean, then it's entirely up to the app what they want to do. I think this approach is reasonable as it's unlikely you're changing id generation policies between environments

Reply:
Yup, tested id, since the code is never reached there is no runtime failure, and the classes aren't lookup, you will only see error if you set the setting for using xray and it isn't in the classpath

Reply:
What if the setting is something other than xray, do you get a ClassNotFound error?

Also, what happens if `TracerUtil` is used without any of the dependencies on the classpath? My understanding would be that the class imports are enough to break `TracerUtil` if the classes are not on the classpath.

Reply:
It doesn't throw class not found, if you set something other than `xray`, we just have this option, it will fail while parsing it to the enum class, other than that it will never get instantiated, and the imports doesn't break it, either way, changed it to use a bean producer =]